x-pack/plugins/lens/public/datatable_visualization/expression.tsx

```diff
@@ -144,23 +189,86 @@ function DatatableComponent(props: DatatableProps & { formatFactory: FormatFacto
         columns={props.args.columns.columnIds
           .map(field => {
             const col = firstTable.columns.find(c => c.id === field);
+            const colIndex = firstTable.columns.findIndex(c => c.id === field);
+
+            const filterable = col?.meta?.type && props.getType(col.meta.type)?.type === 'buckets';
             return {
               field,
               name: (col && col.name) || '',
+              render: (value: unknown) => {
+                const formattedValue = formatters[field]?.convert(value);
+                if (filterable && value) {
```

Comment:
The condition here is preventing filters from being added when the unformatted value is empty string or 0: both of those are valid categories.